components/heading/demo/heading-stack-test.js

```diff
@@ -0,0 +1,41 @@
+import { html, literal } from 'lit/static-html.js';
+import { HeadingLevelController } from '../heading-stack.js';
+import { LitElement } from 'lit';
+
+class TestHeadingStack extends LitElement {
+
+	constructor() {
+		super();
+		this._headingLevelController = new HeadingLevelController(this);
+	}
+
+	render() {
+		let headingLevel;
+		switch (this._headingLevelController.level) {
+			case 1:
+				headingLevel = literal`h1`;
+				break;
+			case 2:
+				headingLevel = literal`h2`;
+				break;
+			case 3:
+				headingLevel = literal`h3`;
+				break;
+			case 4:
+				headingLevel = literal`h4`;
+				break;
+			case 5:
+				headingLevel = literal`h5`;
+				break;
+			case 6:
+				headingLevel = literal`h6`;
+				break;
+		}
```

Comment:
Alternate way to do this:

```javascript
const headingLevel = `h${this._headingLevelController.level}`;
return html`
			<${unsafeStatic(headingLevel)}>heading</${unsafeStatic(headingLevel)}>
			<slot></slot>
`;
```

Reply:
Interesting, so it doesn't let you do the following eh?
```javascript
 literal`h${level}`
```
These aren't a huge deal if they are within our components. The switch statement could be factored away into a helper function that could be shared. Nevertheless, I agree about, and I'd rather consumers not have to be concerns with any of this gnarling. 😬

Reply:
Yeah, the stuff you pass to `literal` has to "literally" be a literal -- no dynamic stuff, unless you use `unsafeStatic` like in the sample above.

I might mock out in this draft what a `<d2l-heading>` component might look like...

Reply:
Yeah... I think I like the idea of the `<d2l-heading>` component - it seems like it would hide all this stuff from the consumer. They would just need to use `<d2l-heading>` and `<d2l-heading-stack>`. I think it could still support all the positioning flexibility that `<h#>` provides, and we'd just have to expose a property for the style level, and perhaps a property for the semantic level so they can override. They'd no longer need to import the heading level styles, so that's a ➕ .

It seems like the viability of this depends on everything on the page using it, ya?

Reply:
For sure yeah, it would have to be all-or-nothing pretty much. Of course, the manually created headings with styles applied would still work in the meantime. I'll continue to flesh this out a bit.